allennlp/data/dataset_readers/squad.py

```diff
@@ -233,8 +231,12 @@ def from_params(cls, params: Params):
         squad_filename = params.pop('squad_filename')
         negative_sentence_selection = params.pop('negative_sentence_selection', 'paragraph')
         tokenizer = Tokenizer.from_params(params.pop('tokenizer', {}))
-        token_indexers = [TokenIndexer.from_params(p)
-                          for p in params.pop('token_indexers', [Params({})])]
+        token_indexers = {}
+        token_indexer_params = params.pop('token_indexers', Params({}))
+        for name, indexer_params in token_indexer_params.items():
+            token_indexers[name] = TokenIndexer.from_params(indexer_params)
+        if token_indexers == {}:
+            token_indexers = None
```

Comment:
And here...  Seems a bit redundant to have it everywhere, but the problem shows up when you read the first file, and we don't know which one you're going to read first.